trie/stacktrie.go

```diff
@@ -429,51 +371,20 @@ func (t *StackTrie) hash(st *stNode, path []byte) {
 	// input values.
 	st.val = t.h.hashData(blob)
 
-	// Short circuit if the stack trie is not configured for writing.
-	if t.options.Writer == nil {
-		return
+	// Invoke the callback it's provided. Notably, the path and blob slices are
+	// volatile, please deep-copy the slices in callback if the contents need
+	// to be saved.
```

Comment:
Please rephrase so it's clear who should do what: 
```suggestion
	// volatile. It is up to the `onTrieNode` receiver function to deep-copy the data if it wants 
	// to retain it after the call ends. 
```
But perhaps that comment should sit up by the definition of the `OnTrieNode` function?

Reply:
> Please rephrase so it's clear who should do what: 
> ```suggestion
> 	// volatile. It is up to the `onTrieNode` receiver function to deep-copy the data if it wants 
> 	// to retain it after the call ends. 
> ```
> But perhaps that comment should sit up by the definition of the `OnTrieNode` function?